tests/cypress/integration/canvas3d_functionality/case_56_canvas3d_functionality_basic_actions.js

```diff
@@ -0,0 +1,158 @@
+// Copyright (C) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+
+/// <reference types="cypress" />
+
+import { taskName } from '../../support/const_canvas3d';
+// Firefox does not yet support WebGL in headless mode: https://bugzilla.mozilla.org/show_bug.cgi?id=1375585 (disabled in the cypress_cron_type.json)
+context('Canvas 3D functionality. Basic actions.', () => {
+    const caseId = '56';
+    const screenshotsPath =
+        'cypress/screenshots/canvas3d_functionality/case_56_canvas3d_functionality_basic_actions.js';
+
+    function compareImages(imgBefore, imgAfter) {
+        cy.compareImages(`${screenshotsPath}/${imgBefore}`, `${screenshotsPath}/${imgAfter}`).then((diffPercent) => {
+            expect(diffPercent).to.be.gt(0);
+        });
+    }
+
+    function checkFilenameWrapperText(expectedText) {
+        cy.get('.cvat-player-filename-wrapper').should('contain.text', expectedText);
+    }
+
+    function pressKeyPerspectiveDirections(key, screenshotNameBefore, screenshotNameAfter) {
+        cy.get('.cvat-canvas3d-perspective').trigger('mouseover').screenshot(screenshotNameBefore);
+        cy.get('body').type(`{alt}${key}`).wait(300); //Wait to change point cloud position
```

Comment:
Shouldn't we use keydown here with corresponding keycode? (as in other tests)
Can we get rid of wait() function? It doesn't look reliable.

Reply:
You don't have to use this for type (). Since this function passes kyeCode. For example ```Details: "{ code: AltLeft, which: 18 }"```
Cypress executes the command too quickly, and without using ```wait()```, the points cloud changes do not have time to occur. In this case, there will be no percentage change in pixels when comparing images.

Reply:
> Cypress executes the command too quickly, and without using wait(), the points cloud changes do not have time to occur. In this case, there will be no percentage change in pixels when comparing images.

Can we wrap the following commands to setTimeout without exact timeout value?

Theoretically event ``cy.get('body').type(`{alt}${key}`)`` generates a macrotask which will be done after current stack execution. If we wrap the next instructions to setTimeout, it also generates a macrotask which will be done after the previous one. 
```
setTimeout(() => {
    cy.get('.cvat-canvas3d-perspective').screenshot(screenshotNameAfter);
    compareImages(`${screenshotNameBefore}.png`, `${screenshotNameAfter}.png`);
})
```

Reply:
In this case, I get an error.
```
Error:                       CypressError: The following error originated from your test code, not from Cypress.
  > Cypress detected that you returned a promise from a command while also invoking one or more cy commands in that promise.
The command that returned the promise was:
  > `cy.screenshot()`
The cy command you invoked inside the promise was:
  > `cy.get()`
Because Cypress commands are already promise-like, you don't need to wrap them or return your own promise.
Cypress will resolve your command with whatever the final Cypress command yields.
The reason this is an error instead of a warning is because Cypress internally queues commands serially whereas Promises execute as soon as they are invoked. Attempting to reconcile this would prevent Cypress from ever resolving.
```
I will try to regenerate the ```.pcd``` files. Perhaps this will help to solve this problem. Since in this view, points cloud is far away in the image and minimal changes are not visible in the screenshot.

Reply:
To exclude the use of ```wait()```, I replaced it with an additional ```click()``` on the element. For additional time, change the orientation of points cloud.

Reply:
Tested it locally and I do not have any issues if I remove both wait() and click(). It is quite strange.